### PR TITLE
Clown shoes can make you waddle

### PR DESCRIPTION
**Pull request by Mickyan**
:cl: Mickyan
tweak: brave clowns can now manually disable the patented Waddle Dampeners(tm) built into their shoes by using ctrl-click.
/:cl:

honk :o)

### CODE REVIEW

**Review by Cyberboss — DISMISSED** [review]

**Comment by Ziiro** [comment]
I don't suppose there's a way to make this something you can toggle by using the shoes in hand before putting them on?

As a regular clown ~~shitter~~ main I still see no problem with it, even if that's not the case. Big 👍

**Comment by kriskog** [comment]
What about chameleon noslips that look like clown shoes? They will be near unusable for stealth without the waddle.

**Comment by Mickyan** [comment]
why must antags ruin everything

I'll make it a toggle that starts off

**Comment by kriskog** [comment]
I liked it better forced. Oh well, you know best

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 1]
```diff
@@ -69,28 +69,44 @@
...
 	if(user.mind && user.mind.assigned_role == "Clown")
 		SEND_SIGNAL(user, COMSIG_CLEAR_MOOD_EVENT, "noshoes")
 
 /obj/item/clothing/shoes/clown_shoes/dropped(mob/user)
 	. = ..()
+	QDEL_NULL(waddle)
 	if(user.mind && user.mind.assigned_role == "Clown")
 		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "noshoes", /datum/mood_event/noshoes)
 
+/obj/item/clothing/shoes/clown_shoes/CtrlClick(mob/living/user)
```
Comment (Cyberboss) [inline code/modules/clothing/shoes/miscellaneous.dm]: I believe any mob can run this, so it'll need ghost checks

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 2]
```diff
@@ -69,28 +69,46 @@
...
 /obj/item/clothing/shoes/clown_shoes
-	desc = "The prankster's standard-issue clowning shoes. Damn, they're huge!"
+	desc = "The prankster's standard-issue clowning shoes. Damn, they're huge! Ctrl-click to toggle waddle dampeners."
 	name = "clown shoes"
 	icon_state = "clown"
 	item_state = "clown_shoes"
 	slowdown = SHOES_SLOWDOWN+1
 	item_color = "clown"
 	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes/clown
+	var/datum/component/waddle
```
Comment (Cyberboss) [inline code/modules/clothing/shoes/miscellaneous.dm]: Holding component references is a bad, use a weakref if at all

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/clothing/shoes/miscellaneous.dm (hunk 2 above)
Comment (Mickyan) [reply code/modules/clothing/shoes/miscellaneous.dm]: I'll need some help with this cause I just copied how it was done for the penguin ears, I'm still learning how to deal with components

File: code/modules/clothing/shoes/miscellaneous.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/clothing/shoes/miscellaneous.dm]: L 100/101 covers this

**Comment by anconfuzedrock** [comment]
Hold on, clownshoes already slow you down, so a noslip disguise is shit anyways for that part exposing you. Probably not a good reason to make it a toggle.

**Comment by ExcessiveUseOfCobblestone** [comment]
fake clown shoes don't squeak either

File: code/modules/clothing/shoes/miscellaneous.dm (hunk 2 above)
Comment (optimumtact) [reply code/modules/clothing/shoes/miscellaneous.dm]: @Cyberboss

Comment (Qustinnus) [reply code/modules/clothing/shoes/miscellaneous.dm]: why is holding component references bad

**Comment by Therapist1** [comment]
merge when

**Pull request merged**